rttanalysis: flakes in BenchmarkVirtualTableQueries

```
[Bench //pkg/bench/rttanalysis:rttanalysis_test] --- FAIL: BenchmarkVirtualTableQueries/virtual_table_cache_with_schema_change
[Bench //pkg/bench/rttanalysis:rttanalysis_test]     test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/1b686e68658fd1761e16df16f8888527/logBenchmarkVirtualTableQueries_virtual_table_cache_with_schema_change525902431
[Bench //pkg/bench/rttanalysis:rttanalysis_test]     test_log_scope.go:79: use -show-logs to present logs inline
[Bench //pkg/bench/rttanalysis:rttanalysis_test]     rtt_analysis_bench.go:175: BenchmarkVirtualTableQueries/virtual_table_cache_with_schema_change: got 9, expected 11
[Bench //pkg/bench/rttanalysis:rttanalysis_test]     rtt_analysis_bench.go:181: wrote jaeger trace to /artifacts/tmp/_tmp/1b686e68658fd1761e16df16f8888527/logBenchmarkVirtualTableQueries_virtual_table_cache_with_schema_change525902431/BenchmarkVirtualTableQueries_virtual_table_cache_with_schema_change.jaeger.json
[Bench //pkg/bench/rttanalysis:rttanalysis_test]     testing_shims.go:42: -- test log scope end --
```

Lots of failures like this, especially in the `release-22.2` branch. Example: <https://teamcity.cockroachdb.com/viewLog.html?buildId=7976461&buildTypeId=Cockroach_Ci_Tests_Bench&tab=buildResultsDiv>

Jira issue: CRDB-22510

--- a/pkg/bench/rttanalysis/testdata/benchmark_expectations
+++ b/pkg/bench/rttanalysis/testdata/benchmark_expectations
@@ -93,4 +93,4 @@ exp,benchmark
 1,VirtualTableQueries/select_crdb_internal.invalid_objects_with_1_fk
 1,VirtualTableQueries/select_crdb_internal.tables_with_1_fk
 4,VirtualTableQueries/virtual_table_cache_with_point_lookups
-11,VirtualTableQueries/virtual_table_cache_with_schema_change
+10,VirtualTableQueries/virtual_table_cache_with_schema_change

